presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -285,6 +309,35 @@ public void addFinalQueryInfoListener(StateChangeListener<QueryInfo> stateChange
         stateMachine.addQueryInfoStateChangeListener(stateChangeListener);
     }
 
+    @Override
+    public Optional<QueryType> getQueryType()
+    {
+        if (statement instanceof Query) {
+            return Optional.of(SELECT);
+        }
+        else if (statement instanceof Explain) {
+            return Optional.of(EXPLAIN);
+        }
+        else if (statement instanceof ShowCatalogs  || statement instanceof ShowCreate || statement instanceof ShowFunctions ||
+                statement instanceof ShowGrants || statement instanceof ShowPartitions || statement instanceof ShowSchemas ||
+                statement instanceof ShowSession || statement instanceof ShowStats || statement instanceof ShowTables) {
+            return Optional.of(SHOW);
+        }
+        else if (statement instanceof CreateTableAsSelect) {
+            return Optional.of(CREATE);
+        }
+        else if (statement instanceof Insert) {
+            return Optional.of(INSERT);
+        }
+        else if (statement instanceof Delete) {
+            return Optional.of(DELETE);
+        }
+        else if (statement instanceof ShowColumns || statement instanceof DescribeInput || statement instanceof DescribeOutput) {
+            return Optional.of(DESCRIBE);
+        }
+        return Optional.empty();
```

Comment:
If I understand correctly, empty basically means other. However, we don't want to use OTHER because that might create backward-compatibility issues in the future. Is that correct?